core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -91,6 +91,19 @@ class NewShinyConsumer(topic: Option[String], partitionId: Option[Int], offset:
     }
   }
 
+  private def seekToRealPositionsBeforeExit() {
+    val smallestUnconsumedOffsets = collection.mutable.Map[TopicPartition, Long]()
+    while (recordIter.hasNext) {
+      val record = recordIter.next()
+      val tp = new TopicPartition(record.topic(), record.partition())
+      // only insert the smallest offset for each partition
```

Comment:
Can you mention that we do this to avoid auto-committing offsets which haven't been returned?